corehq/project_limits/rate_counter/rate_counter.py

```diff
@@ -0,0 +1,156 @@
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import unicode_literals
+from __future__ import division
+
+import hashlib
+import time
+
+import six
+from django.core.cache import caches
+
+from corehq.project_limits.rate_counter.interfaces import AbstractRateCounter
+
+REDIS = caches['default']
+LOCMEM = caches['locmem']
+
+
+class SlidingWindowRateCounter(AbstractRateCounter):
+    """
+    A "Sliding Window Over Fixed Grains" approach that approximates perfect sliding window
```

Comment:
Could delete this line. Its repeated in the next line of comments

Reply:
The [PEP-257 convention for multi-line docstrings](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings) is to have a summary line separate from the rest of the description. Could possibly remove or rephrase the redundant text in the description below, but I think it would be better to keep this.